inception-project/inception
# #5211 - AI assistant prototype

**What's in the PR**
- Initial AI assistant implementation accessible on the annotation page as a sidebar
- Basic access to documents in the project via RAG
- Limited access to user manual via RAG
- Updates to the user manual
- Some tests
- Documentation of the early AI assistant
- Refactoring of the Ollama support
- Fix displaying indexing progress on the bulk processing page
- Added re-index button to assistant sidebar
- Normalize embeddings and use dot product instead of cosine when indexing/searching
- Allow suspending background tasks for a project
- During project import and project initialization from a template, suspend background tasks
- During indexing, embed in batches
- Fix timing logging for embeddings
- Allow auto-detecting the embedding dimensions
- Allow using embeddings with a dimension higher than 1024
- Added button to clear conversation
- Aggregate units (sentences) into larger chunks when indexing
- Extract embedding stuff into a separate service and make tokenizer/encoding configurable
- Implement conversation clearing as protocol command
- Factor retrievers out into own classes and introduce an extension point for them
- Introduce a chunker interface
- Add source information to chunks
- UI style changes
- Some classes renameed
- Added time info in assistant messages
- Fix issue when search index has not been initialized when project is being deleted
- Update settings documentation
- Adjust parameters for embedding
- Lay groundworks for source attribution
- Adjust parameters for embedding
- Allow assistant to provide attribution to document context used in responses
- Allow specifying a document when using the DIAM scrollTo function

**How to test manually**
* Enable the experimental AI assistant
* Make sure Ollama is running and has `llama.3.2` and `granite-embedding` models loaded
* Try talking to it on the annotation page

**Automatic testing**

**Documentation**
